src/store/resources/actions.ts

```diff
@@ -0,0 +1,201 @@
+                cpu_weight: from_others_cpu_asset,
+            };
+
+            store.commit('setDelegatedFromOthers', fromOthers);
+            store.commit('setSelfStaked', selfStaked);
+        } catch (err) {
+            console.error('Error:', err);
+        } finally {
+            store.commit('unsetLoading', 'updateSelfStaked');
+        }
+    },
+
+    // delegated resources actions
+    async updateDelegatedToOthers({ commit }, account: string) {
+        try {
+            commit('setLoading', 'updateDelegatedToOthers');
+
+            const paramsdelband = {
+                code: 'eosio',
+                limit: '200',
+                scope: account,
+                table: 'delband',
+            } as GetTableRowsParams;
+
+            const delegated = (
+                (await api.getTableRows(paramsdelband)) as {
+                    rows: DelegatedResources[];
+                }
+            ).rows;
+
+            commit('setCurrentAccount', account);
+            commit('setDelegatedToOthers', delegated);
+        } catch (err) {
+            console.error('Error:', err);
+        } finally {
+            commit('unsetLoading', 'updateDelegatedToOthers');
+        }
+    },
+    async delegateResources({ commit, dispatch }, order: DelegatedResources) {
+        const { from, to, net_weight, cpu_weight } = order;
+
+        // create two actions, one for the delegatebw and one for the transfer
+        try {
+            void commit('setLoading', 'delegateResources');
+
+            // calculate the total amount
+            const amount = Number(net_weight) + Number(cpu_weight);
+
+            const actions = [
+                {
+                    account: 'eosio',
+                    name: 'delegatebw',
+                    authorization: [
+                        {
+                            actor: from,
+                            permission: 'active',
+                        },
+                    ],
+                    data: {
+                        from,
+                        receiver: to,
+                        stake_net_quantity: net_weight,
+                        stake_cpu_quantity: cpu_weight,
+                        transfer: false,
+                        amount,
+                    },
+                },
+            ];
+            const transaction = (await dispatch('sendTransaction', actions)) as undefined as SignTransactionResponse;
+            commit('setTransaction', transaction.transactionId);
+        } catch (e) {
+            commit('setTransactionError', e);
+        } finally {
+            void commit('unsetLoading', 'delegateResources');
+        }
+    },
+    async undelegateResources({ commit, dispatch }, order: DelegatedResources) {
+        const { from, to, net_weight, cpu_weight } = order;
+
+        // create two actions, one for the delegatebw and one for the transfer
+        try {
+            void commit('setLoading', 'undelegateResources');
```

Comment:
can you explain why you use `void` in a bunch of places? from what i can tell, it serves no purpose. just wondering

Reply:
again, you are right